leftwm-core/src/display_servers/xlib_display_server/event_translate_client_message.rs

```diff
@@ -15,7 +15,10 @@ pub fn from_event(xw: &XWrap, event: xlib::XClientMessageEvent) -> Option<Displa
         let value = event.data.get_long(0);
         match usize::try_from(value) {
             Ok(index) => {
-                return Some(DisplayEvent::SendCommand(Command::GotoTag(index + 1)));
+                return Some(DisplayEvent::SendCommand(Command::GotoTag {
+                    tag: index + 1,
+                    swap: false,
+                }));
```

Comment:
I'm not 100% sure if this should be swap off or swap on.

In the current version on main of leftwm, this is currently swapon by default and dependent on the configuration field `disable_current_tag_swap`. But I have a hunch that this should actually be always swap off. If someone can confirm or infirm my hypothesis it would be great :pray:

Reply:
I would say it should stay swapon to prevent the majority of setups changing.

Reply:
> I would say it should stay swapon to prevent the majority of setups changing.

Do you actually know when/how this code is ran? Or you're just assuming this because the swapping is the default value in config?

Either way it will change the existing setups. But I suspect this swapping here was not intentional.

But as I said, I have a good hunch the swapping here was not introduced intentionally... 

The swapping feature has been introduced in Tue Feb 23 07:23:36 2021 -0500 (a57c91a034a5649531b46bc8d192c58a3d7db277)

And that particular code dates from long time ago, at least Wed Mar 13 16:46:23 2019 -0500 (6d6ee80179c1761a7fa14b6e8536e06639d05d3d)

It's mostly triggered by something in `leftwm-core/src/display_servers/xlib_display_server/event_translate.rs`. It's hard to tell where exactly. Need to investigate more (or someone who knows exactly)

Reply:
> Do you actually know when/how this code is ran?

What do you mean by this? External calls from programs e.g. ```xdotool getwindowfocus windowactivate```?

Reply:
Oh wait I see what you mean now. This is called by external programs like ```xdotool getwindowfocus windowactivate
```. I am really bad at reading today.

Reply:
Which request to the window manager to swap the tags (this one brings the tag with the focused window to the screen). So maybe this should be false.

Reply:
Ahh! xD I knew I had a good hunch. I have no idea what's xdotool, it sounds like something cool though